### PR TITLE
EVG-13538: add option for retryable job

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-13538

Implement retryable options with a flag indicating if the job can retry and how many times it's retried. They're no-ops until [EVG-13540](https://jira.mongodb.org/browse/EVG-13540) is implemented to have a retry handler, but they are serialized so they can be stored in MongoDB.

### CODE REVIEW

File: interface.go [hunk 1]
```diff
@@ -129,6 +136,13 @@ type JobTimeInfo struct {
 	MaxTime    time.Duration `bson:"max_time" json:"max_time,omitempty" yaml:"max_time,omitempty"`
 }
 
+// JobRetryInfo stores configuration and information for a job that can retry.
+// Support for retrying jobs is optional for Queue implementations.
+type JobRetryInfo struct {
```
Comment (julianedwards) [inline interface.go:141]: Why do we need both `Retryable` and `RetryCount`? I think having a single configuration called `RetryCount` or `Retries` would be simpler.

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: interface.go (hunk 1 above)
Comment (Kimchelly) [reply interface.go:141]: `Retryable` tells you whether the job can retry at all. `RetryCount` is not meant to be modified by you - it tells you how many times it's tried in total so far.

Comment (julianedwards) [reply interface.go:141]: Ohh I thought it was like the max retry attempts allowed. Can we set a max?

Comment (Kimchelly) [reply interface.go:141]: [EVG-13544](https://jira.mongodb.org/browse/EVG-13544)

**Review by julianedwards — APPROVED** [review]

**Pull request merged**